### DISCUSSION TITLE
"Recognition of file mtype" errors on signed-linear audio files

### DISCUSSION THREAD

**Discussion by gtjoseph** [rpm-software-management/rpm#2310] [General]
I've got 4 out of almost 1200 sound files in signed-linear format that are causing rpmbuild to fail with messages like `Recognition of file mtype "...calls-waiting-for-rep.sln16" failed: mode 100755 (null)`.  If I run `file` on those 4, they return `Unicode text, UTF-32, big-endian` but there are 9 other files that return the same mime type that don't throw the error.

I believe the issue is either in the rpm dependency checking stuff or libmagic itself and where rpmbuild is getting `mode 100755` from, I'm not sure.  Is that what mode it thinks the files _should_ be?  It's certainly not what mode the file are: 0644.

I've been building this package for years with those same files and not had an issue but I _think_ they may have been warnings before and now are errors.

Currently running 4.18.0 on Fedora 37 and getting the above error.
I just tried with 4.17.1 on Fedora 36 and they show up as warnings but the rpms built successfully.
I'm pretty sure that in even earlier versions, there was no message at all but I can't easily test.

Anyone know what's going on or have suggestions for getting by this?

**Comment by pmatilai** [comment]
That message is just a warning  if the file in question is not executable, so that has changed for whatever reason.

The mode in the message comes from the package, which may inherit it from the disk or be spec defined. This sounds like the spec is accidentally overriding the file mode for these files, but if the same spec builds with older rpm then it sounds there's something stranger going on. Are the specs/srpms exhibiting this available for viewing someplace?

**Reply by gtjoseph** [reply]
I double checked the spec file and it's not messing with the mode and the source filed themselves are set exactly like the other 1200 files.  Let me see if I can work up a reproducable example.
